src/Serilog.Sinks.MongoDB/Sinks/MongoDB/MongoDBSink.cs

```diff
@@ -70,14 +71,15 @@ public class MongoDBSink : PeriodicBatchingSink
             TimeSpan? period = null,
             IFormatProvider formatProvider = null,
             string collectionName = DefaultCollectionName,
-            CreateCollectionOptions collectionCreationOptions = null)
+            CreateCollectionOptions collectionCreationOptions = null,
+            MongoDBJsonFormatter mongoDBJsonFormatter = null)
```

Comment:
Would this still work if we made the parameter of type `ITextFormatter`? This would allow the widest range of formatters to be plugged in, without tying us to the `MongoDBJsonFormatter` type, which needs updating in the future to break its dependency on `JsonFormatter`.

Reply:
I replaced `MongoDBJsonFormatter ` with `ITextFormatter`, although of course the formatter is still expected to produce JSON in order to obtain a BSON document to store in MongoDB